### PR TITLE
Adds saved prayer api changes

### PR DESCRIPTION
**Pull request by cdanwards**
### How do I test this PR?

---

> I am affirming this is my _best_ work ([Ecclesiastes 9:10](https://www.bible.com/bible/97/ECC.9.10.MSG))

## TODO

- [x] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [x] Closes [DEV-XXX](#url)
- [x] No new warnings in tests, in storybook, and in-app
- [ ] Upload GIF(s) of iOS and Android if applicable
- [x] Set two relevant reviewers

## REVIEW

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

---

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

**Comment by redreceipt** [comment]
@IsaacHardy I'm getting an error

Screen Shot 2019-05-01 at 4 42 33 PM

File: packages/apollos-data-prayer/src/prayer-requests/schema.js [hunk 1]
```diff
@@ -19,6 +20,8 @@ const prayerRequestSchema = gql`
     deletePublicPrayerRequest(id: String!): PrayerRequest
     incrementPrayed(id: String!): PrayerRequest
     flagRequest(id: String!): PrayerRequest
+    savePrayer(nodeId: String!): PrayerRequest
```
Comment (redreceipt) [inline packages/apollos-data-prayer/src/prayer-requests/schema.js:23]:
```suggestion
    savePrayer(id: String!): PrayerRequest
```

to match others

File: packages/apollos-data-prayer/src/prayer-requests/schema.js [hunk 2]
```diff
@@ -19,6 +20,8 @@ const prayerRequestSchema = gql`
...
+    unSavePrayer(nodeId: String!): PrayerRequest
```
Comment (redreceipt) [inline packages/apollos-data-prayer/src/prayer-requests/schema.js:24]:
```suggestion
    unSavePrayer(id: String!): PrayerRequest
```

**Review by redreceipt — CHANGES_REQUESTED** [review]

File: packages/apollos-data-prayer/src/prayer-requests/schema.js (hunk 1 above)
Comment (IsaacHardy) [reply packages/apollos-data-prayer/src/prayer-requests/schema.js:23]: I like the name here. A nodeId and different from a regular id.

**Comment by IsaacHardy** [comment]
@redreceipt This shoudl be working now.

File: packages/apollos-data-prayer/src/prayer-requests/schema.js (hunk 1 above)
Comment (redreceipt) [reply packages/apollos-data-prayer/src/prayer-requests/schema.js:23]: a `PrayerRequest` extends Node too, are they not the same thing? Like couldn't we also call the variable for all the prayer queries `nodeId`? Or is the Node for the saved stuff not the same kind of node?

**Comment by redreceipt** [comment]
@IsaacHardy looks good! queries work now but if I run the `savedPrayers` query when I'm not logged in, it just returns an empty array, it should return an authentication error like the other prayer queries.

Screen Shot 2019-05-01 at 8 13 33 PM

File: packages/apollos-data-prayer/src/prayer-requests/schema.js (hunk 1 above)
Comment (IsaacHardy) [reply packages/apollos-data-prayer/src/prayer-requests/schema.js:23]: nodeId is that hashed id we created within the nod apollos framework. Id can mean a number of things.

**Comment by IsaacHardy** [comment]
> @IsaacHardy looks good! queries work now but if I run the `savedPrayers` query when I'm not logged in, it just returns an empty array, it should return an authentication error like the other prayer queries.

Those other queries throw an auth error because we call the auth datasource. In this mutation/query we are doing all of the work in our resolver. So I placed a try/catch to catch errors. Problem is, if the user isn't logged in, the first catch is an error in that datasource. Which should be thrown in that query.

Here's the kicker, if you receive an error in the try/catch, then the returned value is what would be return to the client.

So what I'll do is return the error in that try/catch, but it will never send an auth message.

**Comment by redreceipt** [comment]
can we make the error a little more descriptive? 

Screen Shot 2019-05-01 at 11 10 01 PM

File: packages/apollos-data-prayer/src/prayer-requests/schema.js (hunk 1 above)
Comment (redreceipt) [reply packages/apollos-data-prayer/src/prayer-requests/schema.js:23]: so maybe we rename all the other variables to `nodeID` in the other queries for consistency?

File: packages/apollos-data-prayer/src/prayer-requests/resolver.js [hunk 3]
```diff
@@ -10,6 +10,25 @@ export default {
...
+
+        const stuff = await followings.get();
+        const prayerIds = stuff.map((follow) => follow.entityId);
+        const prayers = await dataSources.PrayerRequest.getFromIds(
+          prayerIds
+        ).get();
+
+        return prayers;
+      } catch (err) {
+        throw new Error(err);
```
Comment (redreceipt) [inline packages/apollos-data-prayer/src/prayer-requests/resolver.js:29]:
```suggestion
        throw new Error(`${err} - Possibly not logged in!`);
```

so it atleast lets them know

Screen Shot 2019-05-01 at 11 26 39 PM

**Review by redreceipt — CHANGES_REQUESTED** [review]

File: packages/apollos-data-prayer/src/prayer-requests/resolver.js (hunk 3 above)
Comment (IsaacHardy) [reply packages/apollos-data-prayer/src/prayer-requests/resolver.js:29]: I don't like that.

Comment (IsaacHardy) [reply packages/apollos-data-prayer/src/prayer-requests/resolver.js:29]: I think we're grasping at straws here. Let's not hold up on error messaging when we have a ton of work to do to launch beta.

**Review by redreceipt — APPROVED** [review]
